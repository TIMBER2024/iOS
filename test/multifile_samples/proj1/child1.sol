pragma solidity "0.6.11";
import {Base} from "./base.sol";
contract Child1 is Base {
    /// #if_succeeds {:msg "P1"} x == old(x) + y;
    function add(uint y) public {
        x += y;
    }
}
