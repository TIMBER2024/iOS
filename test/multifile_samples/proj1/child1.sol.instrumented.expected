pragma solidity 0.6.11;

import { Base } from "./base.sol";
import "./__scribble_ReentrancyUtils.sol";

contract Child1 is __scribble_ReentrancyUtils, Base {
    struct vars2 {
        uint256 old_0;
        bool __scribble_check_invs_at_end;
    }

    function add(uint y) public {
        vars2 memory _v;
        _v.__scribble_check_invs_at_end = __scribble_out_of_contract;
        __scribble_out_of_contract = false;
        _v.old_0 = x;
        _original_Child1_add(y);
        if (!(x == (_v.old_0 + y))) {
            emit AssertionFailed("1: P1");
            assert(false);
        }
        if (_v.__scribble_check_invs_at_end) __scribble_check_state_invariants();
        __scribble_out_of_contract = _v.__scribble_check_invs_at_end;
    }

    function _original_Child1_add(uint y) private {
        x += y;
    }

    /// Check only the current contract's state invariants
    function __scribble_Child1_check_state_invariants_internal() internal {}

    /// Check the state invariant for the current contract and all its bases
    function __scribble_check_state_invariants() virtual override internal {
        __scribble_Child1_check_state_invariants_internal();
        __scribble_Base_check_state_invariants_internal();
    }

    constructor() public {
        __scribble_out_of_contract = false;
        __scribble_check_state_invariants();
        __scribble_out_of_contract = true;
    }
}